### PR TITLE
add async portion, add a logging piece, update tests accordingly

### PR DESCRIPTION
**Pull request by kylepotts**
@danrigsby

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +36,6 @@
     "async": "~1.2.0",
     "gulp-util": "~3.0.5",
     "lodash": "~3.9.3",
-    "through2": "~0.6.5"
+    "through2-concurrent": "^1.1.0"
```
Comment (danrigsby) [inline package.json:40]: Here is the magic sauce

**Comment by danrigsby** [comment]
LGTM

**Comment by kylepotts** [comment]
@andrewall @mjhm

**Comment by kylepotts** [comment]
Should I go ahead and try to make this fit our style guide? Obviously the orginal repo owner had a preference for no semicolons.

**Comment by mjhm** [comment]
If there was any inclination contribute this back to the original `gulp-shell` then I'd keep to the original code style.  However since this looks like a clean break it's probably more appropriate to conform to our style.

**Comment by danrigsby** [comment]
I would not worry about our guidelines here.  I would like to keep the changes as small as possible so that it easy to pull in fixes.

**Comment by kylepotts** [comment]
Updated for code review items and added a test to verify tasks are being run concurrently.

File: index.js [hunk 2]
```diff
@@ -1,11 +1,68 @@
...
+        error: error
+      }, options.templateData)
+
+      error.message = gutil.template(options.errorMessage, errorContext)
+    }
+    if(!options.quiet && options.logMessage){
+      var date = new Date()
+      var hours = date.getHours()
+      var hourStr = hours <= 9 ? "0"+hours : ""+hours
+      var time = hourStr+":"+date.getMinutes()+":"+date.getSeconds()
```
Comment (mjhm) [inline index.js]: You probably wanted to zero pad the minutes and seconds as well.  Here's a cleaner idiom

```
var time = ('0' + hours).slice(-2) + ':' +  ('0' + date.getMinutes()).slice(-2) + ':' +  ('0' + date.getSeconds()).slice(-2);
```

**Comment by mjhm** [comment]
:+1: (except for minor tweak) Good work!

**Pull request merged**